lib/ansible/plugins/connection/httpapi.py

```diff
@@ -264,8 +264,13 @@ def send(self, path, data, **kwargs):
         '''
         Sends the command to the device over api
         '''
+
+        t1 = self.get_option('timeout')
+        t2 = self.get_option('persistent_command_timeout')
+
         url_kwargs = dict(
-            timeout=self.get_option('timeout'), validate_certs=self.get_option('validate_certs'),
+            timeout=t1 if t1 > t2 else t2,
```

Comment:
@nkshrishail We discussed this PR on the call today.  You should be able to remove the `t1`, `t2` and `timeout` references and just add the following line.

```python
timeout=self.get_option('persistent_command_timeout'),
validate_certs=self.get_option('validate_certs'),
...
```

There a couple of different ways to set this timeout that we need to verify.

1. Set the environment variable: `export ANSIBLE_PERSISTENT_COMMAND_TIMEOUT=600`
2. Set it in the `ansible.cfg` file. https://docs.ansible.com/ansible/latest/reference_appendices/config.html#persistent-command-timeout
3. Passed in as a per/task variable. (Example)

```
- name: save running-config
  ios_command:
    commands: copy running-config startup-config
  vars:
    ansible_command_timeout: 60
```

Reply:
@mikewiebe all three options work fine.